cpp/src/arrow/ipc/CMakeLists.txt

```diff
@@ -46,7 +46,14 @@ endfunction()
 
 add_arrow_test(feather_test)
 add_arrow_ipc_test(json_simple_test)
-add_arrow_ipc_test(read_write_test)
+add_arrow_ipc_test(read_write_test
+        SOURCES
+        read_write_test.cc
+        ${CMAKE_CURRENT_SOURCE_DIR}/../filesystem/filesystem.cc
+        ${CMAKE_CURRENT_SOURCE_DIR}/../filesystem/util_internal.cc
+        ${CMAKE_CURRENT_SOURCE_DIR}/../filesystem/path_util.cc
+        ${CMAKE_CURRENT_SOURCE_DIR}/../filesystem/localfs.cc
+        ${CMAKE_CURRENT_SOURCE_DIR}/../filesystem/mockfs.cc)
```

Comment:
I find mockfs API classes are defined in another module `filesystem`, when I use the mockfs API in `ipc` module's `read_write_test.cc`, the linking will fail. I am not very familiar with arrow's cmake organization, and I did some search but I didn't find mockfs usage outside of `filesystem` module. Currently, I add the minimum mockfs related files into arrow_ipc_test to make the linking work. Please let me know if there is better approach doing it.